ARCs/arc-0200.md

```diff
@@ -0,0 +1,232 @@
+        },
+        {
+          "type": "uint256",
+          "name": "value_",
+          "desc": "Amount of tokens to transfer"
+        }
+      ],
+      "returns": { "type": "bool", "desc": "Success" }
+    },
+    {
+      "name": "approve",
+      "desc": "Approve spender for a token",
+      "readonly": false,
+      "args": [
+        { "type": "address", "name": "spender_" },
+        { "type": "uint256", "name": "value_" }
+      ],
+      "returns": { "type": "bool", "desc": "Success" }
+    },
+    {
+      "name": "allowance",
+      "desc": "Returns the current allowance of the spender of the tokens of the owner",
+      "readonly": true,
+      "args": [
+        { "type": "address", "name": "owner_" },
+        { "type": "address", "name": "spender_" }
+      ],
+      "returns": { "type": "uint256", "desc": "The remaining allowance" }
+    }
+  ],
+  "events": [
+    {
+      "name": "Transfer",
+      "desc": "Transfer of tokens",
+      "args": [
+        {
+          "type": "address",
+          "name": "from_",
+          "desc": "The source of transfer of tokens"
+        },
+        {
+          "type": "address",
+          "name": "to_",
+          "desc": "The destination of transfer of tokens"
+        },
+        {
+          "type": "uint256",
+          "name": "value_",
+          "desc": "The amount of tokens transferred"
+        }
+      ]
+    },
+    {
+      "name": "Approval",
+      "desc": "Approval of tokens",
+      "args": [
+        {
+          "type": "address",
+          "name": "owner_",
+          "desc": "The owner of the tokens"
+        },
+        {
+          "type": "address",
+          "name": "spender_",
+          "desc": "The approved spender of tokens"
+        },
+        {
+          "type": "uint256",
+          "name": "value_",
+          "desc": "The amount of tokens approve"
+        }
+      ]
+    }
+  ]
+}
+```
+
+Ownership of a token by a zero address indicates that a token is out of circulation indefinitely, or otherwise burned or destroyed.
+
+The methods `transfer` and `transferFrom` method MUST error when the balance of `from_` is insufficient.
+The `transferFrom` method MUST error unless called by an approved spender as defined by an extension defined in this ARC.
+The methods `transfer` and `transferFrom` MUST emit a `Transfer` event that a transfer is successful.
+A `Transfer` event SHOULD be emitted, with `from_` being the zero address, when a token is first minted.
+A `Transfer` event SHOULD be emitted, with `to_` being the zero address, when a token is destroyed.
+
+The `Approval` event MUST be emitted when the `approve` method is called successfully.
+
+A value of zero for the `approve` method and the `Approval` event indicates no approval.
+When a `Transfer` event emits following the `transferFrom` method, this may also indicate that the approved value for the token is decremented.
+The contract MUST allow multiple operators per owner.
```

Comment:
I think this means that, for example, owner A can approval both account's B and C simultaneously.  Is that right?  Does the MUST here imply an unbounded number of such approvals must be allowed?

Reply:
Yes, there is no limit on approvals for any account.

For example,

approval[A][B] and
approval[A][C] 

are mutually exclusive.

On another note

approval[D][B] and
approval[D][C] 

MUST be allowed

that is there are no bounds on approval relations.

---

> I think this means that, for example, owner A can have an outstanding approval for both account's B and C to spend their tokens simultaneously. Is that right?

Yes, if A has approval for both B and C, B and C can be spent by A simultaneously.

---

Hope that helped